src/Foundation/NSUrl.cs

```diff
@@ -104,18 +112,18 @@ public bool SetResource (NSString nsUrlResourceKey, NSObject value)
 			}
 		}
 
-		public static bool operator == (NSUrl x, NSUrl y)
+		public static bool operator == (NSUrl? x, NSUrl? y)
 		{
-			if ((object) x == (object) y) // If both are null, or both are same instance, return true.
+			if ((object?) x == (object?) y) // If both are null, or both are same instance, return true.
 				return true;
 
-			if (((object) x == null) || ((object) y == null)) // If one is null, but not both, return false.
+			if (((object?) x == null) || ((object?) y == null)) // If one is null, but not both, return false.
```

Comment:
This can be simplified to:

```suggestion
			if (x is null || y is null) // If one is null, but not both, return false.
```